### PR TITLE
Fix epub language

### PR DESCRIPTION
**Pull request by j0k3r**
Ensure language has `_` instead of `-`
And update wallabag/phpepub

Fix https://github.com/wallabag/wallabag/issues/5596

### CODE REVIEW

**Comment by tcitworld** [comment]
> As the comment in the doc says, EPub defined language MUST be on 2 chars.

Are you sure about this? That's not what RFC3066 says. From what I can see ePub references BCP 47 language tags anyway. :thinking:

**Comment by j0k3r** [comment]
At least I've tested locally with Calibre and the generated epub with 2 chars language is properly detected while the one defined with 5 chars language is displayed as English (the default value then).

**Review by Kdecherf — COMMENTED** [review]
EPUB 3.2 Specification states that the language value must be compliant with BCP 47[^1].
BCP 47 redirects to RFC 5646[^2] which specifies a syntax that technically accepts values like "fa-IR".

This sounds more like a bug in the PHPePub library, see https://github.com/wallabag/PHPePub/blob/master/src/PHPePub/Core/EPub.php#L1460

I think we should close this PR and discuss about it on the library side instead.

Edit:
[^1]: https://www.rfc-editor.org/info/bcp47
[^2]: https://www.rfc-editor.org/rfc/rfc5646.txt

**Comment by j0k3r** [comment]
Awesome review @Kdecherf, I'll debug using the lib.

File: src/Wallabag/CoreBundle/Helper/EntriesExport.php [hunk 1]
```diff
@@ -162,7 +162,7 @@ private function produceEpub()
 
         $book->setTitle($this->title);
         // Not needed, but included for the example, Language is mandatory, but EPub defaults to "en". Use RFC3066 Language codes, such as "en", "da", "fr" etc.
-        $book->setLanguage($this->language);
+        $book->setLanguage(substr($this->language, 0, 2));
```
Comment (Kdecherf) [inline src/Wallabag/CoreBundle/Helper/EntriesExport.php:165]: Even though https://github.com/wallabag/PHPePub/pull/3#discussion_r825345271 will fix an issue on PHPePub side, we still need to fix something in wallabag, here is a suggestion @j0k3r as we store language with `_`:

```suggestion
        // EPub specification requires BCP47-compliant languages, thus we replace _ with -
        $book->setLanguage(str_replace('_', '-', $this->language));
```

File: src/Wallabag/CoreBundle/Helper/EntriesExport.php [hunk 2]
```diff
@@ -162,7 +162,8 @@ private function produceEpub()
 
         $book->setTitle($this->title);
         // Not needed, but included for the example, Language is mandatory, but EPub defaults to "en". Use RFC3066 Language codes, such as "en", "da", "fr" etc.
```
Comment (Kdecherf) [inline src/Wallabag/CoreBundle/Helper/EntriesExport.php:164]: Could you remove the first comment?

**Review by Kdecherf — COMMENTED** [review]
Note on a comment to remove, could you also squash your commits?

**Review by Kdecherf — DISMISSED** [review]

**Review by Kdecherf — APPROVED** [review]

**Pull request merged**